plugins/ingest/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -337,12 +353,28 @@ private void setFieldValue(String path, Object value, boolean append) {
         }
     }
 
-    public String getMetadata(MetaData metaData) {
-        return this.metaData.get(metaData.getFieldName());
+    public String getEsMetadata(MetaData esMetadata) {
+        return this.esMetadata.get(esMetadata.getFieldName());
     }
 
-    public void setMetaData(MetaData metaData, String value) {
-        this.metaData.put(metaData.getFieldName(), value);
+    public Map<String, String> getEsMetadata() {
```

Comment:
why do we differentiate between es vs. ingest metadata? why not just have metadata and then we can use namespaces for differentiating between different types of fields (e.g. `es.key` vs. `ingest.key`)?